dav1d.js

```diff
@@ -18,25 +26,46 @@ function getRuntime() {
     maximum: TOTAL_MEMORY / PAGE_SIZE,
   });
   const HEAPU8 = new Uint8Array(memory.buffer);
+  const HEAPU32 = new Uint32Array(memory.buffer);
+  HEAPU32[DYNAMICTOP_PTR >> 2] = DYNAMIC_BASE;
+
   return {
     table: table,
     memory: memory,
-    sbrk: (increment) => {
-      const oldDynamicTop = dynamicTop;
-      dynamicTop += increment;
-      return oldDynamicTop;
-    },
-    emscripten_memcpy_big: (dest, src, num) => {
+    __table_base: 0,
+    __memory_base: 1024,
+    DYNAMICTOP_PTR: DYNAMICTOP_PTR,
+    _emscripten_memcpy_big: (dest, src, num) => {
       HEAPU8.set(HEAPU8.subarray(src, src+num), dest);
     },
+    _emscripten_resize_heap: (requestedSize) => {
+      abort('OOM');
+    },
+    _emscripten_get_heap_size: () => {
+      return memory.buffer.byteLength;
+    },
     // Empty stubs for dav1d.
-    pthread_cond_wait: (cond, mutex) => 0,
-    pthread_cond_signal: (cond) => 0,
-    pthread_cond_destroy: (cond) => 0,
-    pthread_cond_init: (cond, attr) => 0,
-    pthread_cond_broadcast: (cond) => 0,
-    pthread_join: (thread, res) => 0,
-    pthread_create: (thread, attr, func, arg) => 0,
+    _pthread_cond_wait: (cond, mutex) => 0,
+    _pthread_cond_signal: (cond) => 0,
+    _pthread_cond_destroy: (cond) => 0,
+    _pthread_cond_destroy: (cond) => 0,
+    _pthread_cond_init: (cond, attr) => 0,
+    _pthread_cond_broadcast: (cond) => 0,
+    _pthread_join: (thread, res) => 0,
+    _pthread_create: (thread, attr, func, arg) => 0,
+    _pthread_attr_init: (attr) => 0,
+    _pthread_attr_destroy: (attr) => 0,
+    _pthread_attr_setstacksize: (attr, stacksize) => 0,
+    _abort: abort,
+    abort: abort,
+    ___setErrNo: (value) => {
+      HEAPU32[wasmModule["___errno_location"]() >> 2] = value
+    },
+    ___syscall6: () => { abort('syscall6'); },
```

Comment:
What code requires those syscalls? (close, llseek, writev) I think it's better to update dav1d's patch to avoid extra code to be generated.